cmd/osv-scanner/main.go

```diff
@@ -168,9 +171,10 @@ func run(args []string, stdout, stderr io.Writer) int {
 				NoIgnore:             context.Bool("no-ignore"),
 				ConfigOverridePath:   context.String("config"),
 				DirectoryPaths:       context.Args().Slice(),
+				NoCallAnalysis:       context.StringSlice("no-call-analysis"),
+				CallAnalysis:         context.StringSlice("call-analysis"),
```

Comment:
I think the library should just take a map of bools and then use the default for a language if its not set in the map; how the flags are structured as an implementation detail of the CLI that doesn't need to be inflicted on the library

Reply:
Updated. I moved the createCallAnalysisStates function a new file to keep main.go less cluttered